include/oneapi/dpl/experimental/kt/esimd_radix_sort_submitters.h

```diff
@@ -0,0 +1,253 @@
+            [&](sycl::handler& __cgh)
+            {
+                oneapi::dpl::__ranges::__require_access(__cgh, __keys_tmp_rng, __keys_rng);
+                // TODO: make sure that access is read_only for __keys_tmp_rng  and is write_only for __keys_rng
+                auto __tmp_data = __keys_tmp_rng.data();
+                auto __out_data = __keys_rng.data();
+                __cgh.depends_on(__e);
+                __cgh.parallel_for<_Name...>(sycl::range<1>{__n},
+                                             [=](sycl::item<1> __item)
+                                             {
+                                                 auto __global_id = __item.get_linear_id();
+                                                 __out_data[__global_id] = __tmp_data[__global_id];
+                                             });
+            });
+    }
+};
+
+template <bool __is_ascending, ::std::uint8_t __radix_bits, ::std::uint16_t __data_per_work_item,
+          ::std::uint16_t __work_group_size, typename _KeyT, typename _ValT, typename _KernelName>
+struct __radix_sort_onesweep_by_key_submitter;
+
+template <bool __is_ascending, ::std::uint8_t __radix_bits, ::std::uint16_t __data_per_work_item,
+          ::std::uint16_t __work_group_size, typename _KeyT, typename _ValT, typename... _Name>
+struct __radix_sort_onesweep_by_key_submitter<__is_ascending, __radix_bits, __data_per_work_item, __work_group_size, _KeyT, _ValT,
+                                       oneapi::dpl::__par_backend_hetero::__internal::__optional_kernel_name<_Name...>>
+{
+    template <typename _InKeysRng, typename _OutKeysRng, typename _InValsRng, typename _OutValsRng, typename _GlobalHistT>
+    sycl::event
+    operator()(sycl::queue& __q, _InKeysRng& __in_keys_rng, _OutKeysRng& __out_keys_rng, _InValsRng& __in_vals_rng, _OutValsRng& __out_vals_rng, _GlobalHistT* __p_global_hist, _GlobalHistT* __p_group_hists,
+               ::std::uint32_t __sweep_work_group_count, ::std::size_t __n, ::std::uint32_t __stage,
+               const sycl::event& __e) const
+    {
+        sycl::nd_range<1> __nd_range(__sweep_work_group_count * __work_group_size, __work_group_size);
+        return __q.submit(
+            [&](sycl::handler& __cgh)
+            {
+                oneapi::dpl::__ranges::__require_access(__cgh, __in_keys_rng, __out_keys_rng, __in_vals_rng, __out_vals_rng);
+                auto __in_keys_data = __in_keys_rng.data();
+                auto __out_keys_data = __out_keys_rng.data();
+                auto __in_vals_data = __in_vals_rng.data();
+                auto __out_vals_data = __out_vals_rng.data();
+                __cgh.depends_on(__e);
+                __radix_sort_onesweep_by_key_slm_reorder_kernel<__is_ascending, __radix_bits, __data_per_work_item, __work_group_size,
+                                                       _KeyT, _ValT, decltype(__in_keys_data), decltype(__out_keys_data), decltype(__in_vals_data), decltype(__out_vals_data)>
```

Comment:
`decltype(__in_keys_data), decltype(__out_keys_data), decltype(__in_vals_data), decltype(__out_vals_data)`
Do we really need declare this types?
If we skip all of them, will they deduced automatically?

Reply:
Indeed, it is not required. This approach is derived from the existing key sort implementation. I preserved it with the intention of enhancing code quality synchronously for both implementation in a separate PR.